### PR TITLE
indegree(), outdegree() - accept array of strings

### PR DESCRIPTION
**Pull request by nafraf**
These changes are to implement #2781

Strings and array of strings are accepted as arguments:
```
GRAPH.QUERY g "CREATE (a:A)-[:R]->(b:B)"

127.0.0.1:6379> GRAPH.QUERY g "MATCH (a:A) RETURN outdegree(a, 'R', ['R','R'])"
1) 1) "outdegree(a, 'R', ['R','R'])"
2) 1) 1) (integer) 1

127.0.0.1:6379> GRAPH.QUERY g "MATCH (a:A) RETURN outdegree(a, ['R','R'],'R')"
1) 1) "outdegree(a, ['R','R'],'R')"
2) 1) 1) (integer) 1
```

### CODE REVIEW

**Comment by swilly22** [comment]
@LiorKogan, @nafraf 
we would like to support only the following two signatures:
1. `outdegree(NODE, ARRAY_OF_STRINGS)`
2. `outdegree(NODE, STR_0, STR_1, ... STR_N)`

Either a single array of strings is provided or a variable number of individual strings.
I won't want to support a mixture of the two.

File: src/datatypes/array.h [hunk 1]
```diff
@@ -49,6 +49,14 @@ u_int32_t SIArray_Length(SIValue siarray);
   */
 bool SIArray_ContainsType(SIValue siarray, SIType t);
 
+/**
+  * @brief  Returns true if the array contains an element equals to 'value'
+  * @param  siarray: array
+  * @param  value: value to search for
+  * @retval a boolean indicating whether value was found in siarray
+  */
+bool SIArray_ContainsValue(SIValue siarray, SIValue value);
```
Comment (swilly22) [inline src/datatypes/array.h:58]: Update `SIValue AR_IN(SIValue *argv, int argc, void *private_data) {` `list_funcs.c` to use this new function.

File: src/datatypes/array.c [hunk 2]
```diff
@@ -50,6 +50,18 @@ bool SIArray_ContainsType(SIValue siarray, SIType t) {
 	return false;
 }
 
+bool SIArray_ContainsValue(SIValue siarray, SIValue value) {
```
Comment (swilly22) [inline src/datatypes/array.c:53]: Please duplicate comments from header file to C file

File: src/datatypes/array.c [hunk 3]
```diff
@@ -50,6 +50,18 @@ bool SIArray_ContainsType(SIValue siarray, SIType t) {
...
+	uint array_len = SIArray_Length(siarray);
+	for(uint i = 0; i < array_len; i++) {
+		SIValue elem = SIArray_Get(siarray, i);
```
Comment (swilly22) [inline src/datatypes/array.c:56]: Although this is a good practice to use the API, I think that while in the same C file we can access the datatype internals (which is a bit cheaper) and work with the underline array
```c
SIValue elem = siarray.array[i];
```

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: src/datatypes/array.h (hunk 1 above)
Comment (LiorKogan) [reply src/datatypes/array.h:58]: @swilly22 I believe other combinations are already supported and tested. Do you really think we should put in extra work now just to limit it?

File: src/datatypes/array.c (hunk 2 above)
Comment (nafraf) [reply src/datatypes/array.c]: Ok. Done in [aa6194f](https://github.com/RedisGraph/RedisGraph/pull/2807/commits/aa6194fbec956a35195dd72b6149b5636e2208a7)

File: src/datatypes/array.c (hunk 3 above)
Comment (nafraf) [reply src/datatypes/array.c]: Ok. Change done in [aa6194f](https://github.com/RedisGraph/RedisGraph/pull/2807/commits/aa6194fbec956a35195dd72b6149b5636e2208a7)

**Comment by nafraf** [comment]
@swilly22 @LiorKogan 
I implemented the validations to limit to two signatures.
These are the messages in case of error:
```
127.0.0.1:6379> GRAPH.QUERY g "MATCH (a:A) RETURN outdegree(a, 'R', ['R'])"
(error) Type mismatch: expected String but was List
```
```
127.0.0.1:6379> GRAPH.QUERY g "MATCH (a:A) RETURN outdegree(a, ['R'], 'R')"
(error) Received 3 arguments, expected at most 2 because second argument is List
```

**Pull request merged**